ui/apps/platform/src/Containers/Integrations/IntegrationTilesPage/IntegrationTilesPage.tsx

```diff
@@ -1,70 +1,104 @@
-import React from 'react';
-import PropTypes from 'prop-types';
-import { connect } from 'react-redux';
+import React, { ReactElement } from 'react';
+import { useSelector } from 'react-redux';
 import { createStructuredSelector } from 'reselect';
 import { PageSection, Title } from '@patternfly/react-core';
 
 import useCentralCapabilities from 'hooks/useCentralCapabilities';
 import useFeatureFlags from 'hooks/useFeatureFlags';
 import { integrationsPath } from 'routePaths';
 import { selectors } from 'reducers';
-import integrationsList from '../utils/integrationsList';
+import { ClusterInitBundle } from 'services/ClustersService';
+import { ApiToken } from 'types/apiToken.proto';
+import { BackupIntegration } from 'types/externalBackup.proto';
+import { ImageIntegration } from 'types/imageIntegration.proto';
+import { NotifierIntegration } from 'types/notifier.proto';
+import { SignatureIntegration } from 'types/signatureIntegration.proto';
+
+import integrationsList, { IntegrationDescriptor } from '../utils/integrationsList';
 
 import IntegrationTile from './IntegrationTile';
 import IntegrationsSection from './IntegrationsSection';
 
-const IntegrationTilesPage = ({
-    apiTokens,
-    clusterInitBundles,
-    backups,
-    imageIntegrations,
-    notifiers,
-    signatureIntegrations,
-}) => {
+type IntegrationSource = keyof typeof integrationsList;
+
+// Although unnecessary for reducers in TypeScript, we intend to replace reducers and sagas with requests.
```

Comment:
I'm not sure what this comment means?

Reply:
Good question.

Because TypeScript can infer types if reducers are written in TypeScript, we would rewrite the reducers, but  work-around here until we can remove reducers and sagas.